src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_sdpa_fusion.cpp

```diff
@@ -53,7 +53,20 @@ StatefulSDPAFusion::StatefulSDPAFusion() {
     auto concat_k = makePattern<opset1::Concat>({gather_input_k, cur_k}, {{"axis", axis_seq_len}});
     auto concat_v = makePattern<opset1::Concat>({gather_input_v, cur_v}, {{"axis", axis_seq_len}});
 
-    auto multi_query_bcst = [](std::shared_ptr<Node> kv) {
+    auto multi_query_bcst_bf16 = [](std::shared_ptr<Node> kv) {
+        auto reshape_kv = wrap_type<opset6::Reshape>({kv, any_input()});
+        auto unsqueeze_kv = makePattern<opset1::Unsqueeze>({kv, -2});
+        auto constant_bcst = makeConst(ov::element::bf16, ov::PartialShape("[...]"), [](ov::op::v0::Constant& node) {
+            const auto& bcst_arg = node.cast_vector<ov::bfloat16>();
```

Comment:
Can we merge multi_query_bcst_bf16 and multi_query_bcst_fp32? Some way like -
```suggestion
        auto constant_bcst = makeConst(kv->get_element_type(), ov::PartialShape("[...]"), [](ov::op::v0::Constant& node) {
``` 
and for ov::bfloat16, to use some kind of mapping, maybe.

Reply:
In the pattern creation phase, the element type is unknow, kv->get_element_type() only returns element::dynamic which is useless. If we want to reduce duplicate code the only way here is to share the lambda function add a elementType as its args.

Reply:
another option might be to use an Any node here, and check if it is a Const with desired data type in the callback. 

interesting.. why it is a dynamic element type here. somewhere a bug?

Reply:
It's not a bug, the nodes defined in pattern are type-less, in that phase, nodes has no precision info available. The check of type could only be done later after the pattern is matched.

Reply:
fix in 6ca10941f2955bb8be335626dcc9420a0af7faa5